tests/test_help_messages.py

```diff
@@ -4,20 +4,63 @@
 )
 
 
-def iter_through_all_commands_paths():
+def iter_through_all_commands_and_subcommands_paths():
     """
     Generator iterating through all commands.
     Paths are yielded as List[str]
     """
     ignore_plugins = ["render"]
 
+    no_command = []
+    yield no_command
+
     def _iter_through_commands(command, path):
         yield list(path)
         for subpath, subcommand in getattr(command, "commands", {}).items():
             path.append(subpath)
             yield from _iter_through_commands(subcommand, path)
             path.pop()
 
+    builtin_plugins = load_only_builtin_command_plugins()
+    for plugin in builtin_plugins:
+        spec = plugin.command_spec
+        if not plugin.plugin_name in ignore_plugins:
+            yield from _iter_through_commands(
+                spec.command, spec.full_command_path.path_segments
+            )
+
+
+@pytest.mark.parametrize(
+    "command",
+    iter_through_all_commands_and_subcommands_paths(),
+    ids=(".".join(cmd) for cmd in iter_through_all_commands_and_subcommands_paths()),
+)
+def test_help_messages_invocation(runner, snapshot, command):
+    """
+    Check help messages against the snapshot
+    """
+    result = runner.invoke(command + ["--help"])
+    assert result.exit_code == 0
+    assert result.output == snapshot
+
+
+def iter_through_all_commands_paths():
+    """
+    Generator iterating through all commands.
+    Paths are yielded as List[str]
+    """
+    ignore_plugins = ["render"]
+
+    no_command = []
+    yield no_command
+
+    def _iter_through_commands(command, path):
+        yield list(path)
+        if not plugin.plugin_name in ignore_plugins:
+            yield from _iter_through_commands(
+                spec.command, spec.full_command_path.path_segments
```

Comment:
This falls into an infinite loop, did you mean `spec.full_command_path.path_segments + [subcommand]`?